### DISCUSSION TITLE
What rendering errors does handleError in hooks.client.js catch?

### DISCUSSION THREAD

**Discussion by EmilTholin** [sveltejs/kit#8294] [Q&A]
[This interesting question was asked in the Discord chat](https://discord.com/channels/457912077277855764/1058083586068197446).

This following setup throws an error that is caught by `handleError` in `hooks.client.js`:

```js
// src/hooks.client.js
/** @type {import('@sveltejs/kit').HandleClientError} */
export function handleError({ error, event }) {
    console.log('Client-side handle error!');

    return {
        message: 'Whoops!'
    };
}
```
```js
// src/routes/+page.js
import { browser } from '$app/environment';

export function load() {
    if (browser) throw new Error('Ouch!');

    return {};
}
```
```svelte
<!-- src/routes/+page.svelte -->
<script>
  export let data;
</script>
```

If we remove the error in `load` above and change `+page.svelte` to the following instead...

```svelte
<!-- src/routes/+page.svelte -->
<script>
  export let data;

  if (typeof window !== 'undefined') throw new Error('Man!');
</script>
```

... the error is _not_ caught by `handleError` in `hooks.client.js`.

[The documentation states that](https://kit.svelte.dev/docs/hooks#shared-hooks-handleerror) "If an unexpected error is thrown during loading or rendering, this function will be called...". Am I misunderstanding something about how `handleError` works in the client, or should perhaps the documentation be rephrased somewhat?

**Comment by technopahadi** [comment]
This should be answered or docs be fixed (if no such case)!

**Comment by Pensarfeo** [comment]
I have the same issue!
